### DISCUSSION TITLE
namespace gfx

### DISCUSSION THREAD

**Discussion by Morwenn** [timsort/cpp-TimSort#39] [General]
I just realized that the namespace `gfx` has two million hits on GitHub alone: it is most likely a common abbreviation for "graphics" and it's used in big engines (it notably appears in the Fuchsia source code).

I'm not sure what to do about this. We could probably play it safe and use `namespace timsort` instead, which apparently doesn't exist. That would give a `timsort::timsort(...)` function, a `timsort/timsort.hpp` header to include and a `timsort::timsort` CMake target. It doesn't feel like the coolest solution, but it's probably pragmatic enough.

**Comment by vedgy** [comment]
Not sure if the `timsort` namespace is a good idea. The namespace name clash is relatively benign. An entity other than namespace could be named `timsort` and *that* conflict would not compile... If you decide that such a clash is unlikely, then consider the `timsort::sort` and `timsort::merge` names rather than the repetitive `timsort::timsort` and `timsort::timmerge`.

**Comment by Morwenn** [comment]
To be honest I don't plan to change anything until someone complains that it breaks things in their code base, just letting the issue opened to let people know that we're aware of the potential issue.
